src/OutputWriters/output_construction.jl

```diff
@@ -1,14 +1,21 @@
-using Oceananigans.Fields: validate_indices, Reduction
+using Oceananigans.Fields: validate_indices, Reduction, indices
 using Oceananigans.AbstractOperations: AbstractOperation, ComputedField
 using Oceananigans.Grids: default_indices
 
-restrict_to_interior(::Colon, loc, topo, N) = interior_indices(loc, topo, N)
-restrict_to_interior(::Colon, ::Nothing, topo, N) = UnitRange(1, 1)
-restrict_to_interior(index::UnitRange, ::Nothing, topo, N) = UnitRange(1, 1)
+restrict_to_interior(::Colon, ::Colon, loc, topo, N) = interior_indices(loc, topo, N)
```

Comment:
here we can probably just return `Colon`?

Reply:
No, basically this returns either `1:N` (for center fields) or `1:N+1` (for face fields) --- the range of interior points.

This convenience feature is a little annoying. Basically the user can control index ranges through two arguments. One specifies indices. The other specifies whether to chop halos or not. If the user askes for `index::Colon` but no halos, we have to return something like `1:N` when the field covers all indices. The complication addressed in this PR is when the available indices are restricted (ie the output field is windowed)

Reply:
Ok, (maybe not in this PR) but I wonder if we can make `with_halo=true` the default. An example of where this "restricing to the interior" might have unintended (or silent) consequences is when we try to compute derivatives on post-processed fields and find out that we cannot express derivatives across the whole field but we are restricted to one point less because the field has been effectively chopped off to only the interior silently (I am guilty of something like this happening to me). This seems more like a behavior we want to explicitly decide to have rather than it being implicitly enforced as a default.